From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 14 Jan 2022 09:11:19 -0500
Subject: [PATCH] Rename hackrf.dll to hackrf-0.dll on Windows.

This allows installing multiple major versions of the library on
Windows, which brings feature parity to the Unix builds for packaging
into a libhackrf0 package.
---
 host/libhackrf/src/CMakeLists.txt | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/host/libhackrf/src/CMakeLists.txt b/host/libhackrf/src/CMakeLists.txt
index 9d54a8c..b7bd71c 100644
--- a/host/libhackrf/src/CMakeLists.txt
+++ b/host/libhackrf/src/CMakeLists.txt
@@ -2,18 +2,18 @@
 # Copyright (c) 2012, Jared Boone <jared@sharebrained.com>
 # Copyright (c) 2013, Benjamin Vernoux <titanmkd@gmail.com>
 # Copyright (c) 2013, Michael Ossmann <mike@ossmann.com>
-# 
+#
 # All rights reserved.
-# 
+#
 # Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
-# 
+#
 #     Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
-#     Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the 
+#     Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the
 # 	documentation and/or other materials provided with the distribution.
 #     Neither the name of Great Scott Gadgets nor the names of its contributors may be used to endorse or promote products derived from this software
 # 	without specific prior written permission.
-# 
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, 
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
 # THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
 # IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 # (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
@@ -30,6 +30,10 @@ set(c_headers ${CMAKE_CURRENT_SOURCE_DIR}/hackrf.h CACHE INTERNAL "List of C hea
 add_library(hackrf SHARED ${c_sources})
 set_target_properties(hackrf PROPERTIES VERSION ${MAJOR_VERSION}.${MINOR_VERSION}.0 SOVERSION 0)
 
+if(MSVC)
+    set_target_properties(hackrf PROPERTIES RUNTIME_OUTPUT_NAME "hackrf-${MAJOR_VERSION}")
+endif()
+
 # Static library
 add_library(hackrf-static STATIC ${c_sources})
 if(MSVC)
@@ -43,7 +47,7 @@ set_target_properties(hackrf-static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 
 # Dependencies
 target_link_libraries(hackrf ${LIBUSB_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-   
+
 # For cygwin just force UNIX OFF and WIN32 ON
 if( ${CYGWIN} )
   SET(UNIX OFF)
-- 
2.34.1

